presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveMixedCaseSupport.java

```diff
@@ -0,0 +1,218 @@
+        query("CREATE TABLE " + SCHEMA_NAME_UPPER + ".TESTTABLE3 (name VARCHAR(50), id INT, num DOUBLE)");
+
+        // Verify table existence and column properties
+        assertThat(query("SHOW TABLES FROM " + SCHEMA_NAME))
+                .containsOnly(row("testtable0"), row("testtable"), row("testtable1"), row("testtable2"));
+
+        assertThat(query("SHOW TABLES FROM " + SCHEMA_NAME_UPPER))
+                .containsOnly(row("testtable4"), row("testtable02"), row("testtable3"));
+
+        assertThat(query("DESCRIBE " + SCHEMA_NAME + ".testtable0"))
+                .contains(row("name", "varchar(50)", "", ""), row("id", "integer", "", ""));
+
+        assertThat(query("DESCRIBE " + SCHEMA_NAME + ".testtable"))
+                .contains(row("name", "varchar(50)", "", ""), row("id", "integer", "", ""));
+
+        assertThat(query("DESCRIBE " + SCHEMA_NAME_UPPER + ".testtable4"))
+                .contains(row("name", "varchar(50)", "", ""), row("id", "integer", "", ""));
+
+        assertThat(query("DESCRIBE " + SCHEMA_NAME_UPPER + ".testtable02"))
+                .contains(row("name", "varchar(50)", "", ""), row("id", "integer", "", ""), row("num", "double", "", ""));
+    }
+
+    /*
+     * This test validates inserting data into tables with different case variations in schema and table names.
+     * It ensures that data is inserted and retrieved correctly regardless of case sensitivity.
+     */
+    @Test(groups = {MIXED_CASE})
+    public void testInsertDataWithMixedCaseNames()
+    {
+        // Insert data into tables with different case variations
+        query("INSERT INTO " + SCHEMA_NAME + ".testtable VALUES ('amy', 112), ('mia', 123)");
+        query("INSERT INTO " + SCHEMA_NAME + ".TESTTABLE2 VALUES ('ann', 112), ('mary', 123)");
+        query("INSERT INTO " + SCHEMA_NAME_UPPER + ".testtable02 VALUES ('emma', 112, 200.002), ('mark', 123, 300.003)");
+        query("INSERT INTO " + SCHEMA_NAME_UPPER + ".TESTTABLE3 VALUES ('emma', 112, 200.002), ('mark', 123, 300.003)");
+
+        // Validate inserted data
+        assertThat(query("SELECT * FROM " + SCHEMA_NAME + ".testtable"))
+                .containsOnly(row("amy", 112), row("mia", 123));
+        assertThat(query("SELECT * FROM " + SCHEMA_NAME + ".TESTTABLE2"))
+                .containsOnly(row("ann", 112), row("mary", 123));
+        assertThat(query("SELECT * FROM " + SCHEMA_NAME_UPPER + ".testtable02"))
+                .containsOnly(row("emma", 112, 200.002), row("mark", 123, 300.003));
+        assertThat(query("SELECT * FROM " + SCHEMA_NAME_UPPER + ".TESTTABLE3"))
+                .containsOnly(row("emma", 112, 200.002), row("mark", 123, 300.003));
+    }
+
+    /*
+     * This test verifies selecting data from tables with different case variations in schema and table names.
+     * It ensures that queries return correct results regardless of case sensitivity.
+     */
+    @Test(groups = {MIXED_CASE})
+    public void testSelectDataWithMixedCaseNames()
```

Comment:
This test isn't really any different than the above test. I also think it is not necessary to check on the data path.

Reply:
Yes, SELECT looks the same as in the INSERT one. So I think we can remove this one altogether and just keep testInsertDataWithMixedCaseNames() one as it is. WDYT?